modules/tensor_mechanics/src/actions/InclinedNoDisplacementBCAction.C

```diff
@@ -0,0 +1,100 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InclinedNoDisplacementBCAction.h"
+#include "Factory.h"
+#include "FEProblem.h"
+#include "Conversion.h"
+
+registerMooseAction("TensorMechanicsApp", InclinedNoDisplacementBCAction, "add_bc");
+
+template <>
+InputParameters
+validParams<InclinedNoDisplacementBCAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("Set up inclined no displacement boundary conditions");
+
+  params.addRequiredParam<std::vector<BoundaryName>>(
+      "boundary", "The list of boundary IDs from the mesh where the pressure will be applied");
+
+  params.addParam<VariableName>("disp_x", "The x displacement");
```

Comment:
Get rid of the independent `disp_x`, `disp_y`, and `disp_z` parameters. That's the old way of doing this, and we're getting rid of it.

Reply:
done